### ISSUE TITLE
[SOLVED] 3 equal sites, 2 different search results (attributes not filtering) - how can I debug this?

### ISSUE THREAD

**Issue by pedlima** [10up/ElasticPress#2698]
Guys, we are getting crazy over here.

We lost already so many hours and days looking into this and couldn't find a reason for this so far.

So all these websites have the same basis code:
https://americana.pt/loja/?filter_marca=agatha-ruiz-la-prada
https://revenda.americana.pt/loja/?filter_marca=agatha-ruiz-la-prada
https://pro.americana.pt/loja/?filter_marca=agatha-ruiz-la-prada

But when we do that EXACT search, only the first one gives correct results, the others do not!

We already deleted index and reindexed many times and it does not work. Settings, weights, etc. Everything is exactly the same on the 3 websites.

We have debugged and found no evidences.

How can we get over this and find the solution for this problem?

Any help will be much appreciated.

**Comment by pedlima** [comment]
Ok, just found out a difference between those two websites with attribute filters not working and the one site working properly.

In the two that are wrong, in my Query Args, the post__not_in have information whether in the one website working correctly, that one is empty.

What can be causing that? Is that the problem for all this?

**Comment by pedlima** [comment]
Ok, yet another weird discovery!

Here is part of the "Query Args" from the first website (the one that works):

```
["tax_query"]=>
  array(3) {
    ["relation"]=>
    string(3) "AND"
    [0]=>
    array(5) {
      ["taxonomy"]=>
      string(8) "pa_marca"
      ["field"]=>
      string(4) "slug"
      ["terms"]=>
      array(1) {
        [0]=>
        string(20) "agatha-ruiz-la-prada"
      }
      ["operator"]=>
      string(2) "IN"
      ["include_children"]=>
      bool(false)
    }
    [1]=>
    array(4) {
      ["taxonomy"]=>
      string(18) "product_visibility"
      ["field"]=>
      string(16) "term_taxonomy_id"
      ["terms"]=>
      array(1) {
        [0]=>
        int(7)
      }
      ["operator"]=>
      string(6) "NOT IN"
    }
  }
```

And here's that same part on both the other websites that are NOT filtering the attributes:

```
["tax_query"]=>
  array(2) {
    ["relation"]=>
    string(3) "AND"
    [0]=>
    array(4) {
      ["taxonomy"]=>
      string(18) "product_visibility"
      ["field"]=>
      string(16) "term_taxonomy_id"
      ["terms"]=>
      array(1) {
        [0]=>
        int(7)
      }
      ["operator"]=>
      string(6) "NOT IN"
    }
  }
```

So here is surely the problem. Now, how can this be happening if code is the SAME on the three websites? This is mindblowing!

**Comment by pedlima** [comment]
...

Whoever reads this, won't believe it but this had something to do with our database!

I mean, it was not possible to be any other thing rather than that because everyhing was the same, regarding code.

So conclusion:

In WP Admin, we went to WooCommerce > Status > Tools and then we cleaned some things and regenerated other things.

In the end, that's what made it work again...

The weirdest thing is that when we deactivated EP, the search worked!

Never seen anything like this and me and my team work for many years with WP and have developed some complex $hits... unbelievable how a very basic solution can be the cause of so many problems.
